api/src/main/java/ai/djl/Device.java

```diff
@@ -53,12 +53,22 @@ private Device(String deviceType, int deviceId) {
      *
      * @param deviceType the device type, typically CPU or GPU
      * @param deviceId the deviceId on the hardware.
-     * @return a {@code Device} instance
+     * @return a {@code Device} instance, null if not available
      */
     public static Device of(String deviceType, int deviceId) {
         if (Type.CPU.equals(deviceType)) {
             return CPU;
         }
+        if (Type.GPU.equals(deviceType)) {
+            if (getGpuCount() == 0) {
```

Comment:
getGpuCount() function is not free, it need make a JNA call to cuda driver, we can move into CACHE.computeIfAbsent().

I'm not a fan of returning null in this case, I think throwing IllegalArgumenetException is a better choice. User should check GPU availability before creating device